MLOps/intro-to-ml-model-deployment-and-management/intro-to-ml-model-deployment-and-management.md

```diff
@@ -0,0 +1,1010 @@
+    - Scorer: scorer metric used for this model
+    - Score: validation score using this model
+    - Owner: user who deployed this model
+    - Environment: current environment this model is deployed to
+    - Tag: tag name and value associated with this model deployment
+- Save and deploy
+
+4\. Press the **X** button in the top right corner to exit that **Deployment** page for **hydraulic_model_1**.
+
+### Interactive Scoring via Dev Deployment Environment
+
+1\. Click on **Actions 3 dot** for **hydraulic_model_1 model deployments**, then click **Show sample request**:
+
+![model-deployment-show-sample-request](./assets/model-deployment-show-sample-request.jpg)
+
+**Figure:** Model Deployment Show Sample Request
+
+Click **Copy request** button.
+
+Click **Close** button.
+
+Paste **Sample Request** into your terminal:
+
+![sample-request-result](./assets/sample-request-result.jpg)
+
+**Figure:** Model Deployment Sample Request Result
+
+### Batch Scoring via Dev Deployment Environment
+
+From Driverless AI, we insert live code for Data Recipe on our hydraulic_test data and send it as a score request to MLOps Model deployed in Dev environment.
+
+1\. Go to **Driverless AI** tab
+
+2\. Go to Driverless AI **Datasets** page
+
+3\. Click on **hydraulic_test_data**
+
+4\. Click **Details**
+
+5\. Click **Modify by Recipe** and then click **Live Code**
+
+![dai-test-data-recipe-live-code](./assets/dai-test-data-recipe-live-code.jpg)
+
+6\. Erase the previous live code template
+
+![dai-test-data-initial-live-code](./assets/dai-test-data-initial-live-code.jpg)
+
+7\. Copy the following **Data Pumper Python code** into Driverless AI **live code** editor:
+
+8\. Update the `scoring_endpoint` variable value with your particular scoring_endpoint for your current deployed model.
+
+~~~python
+import csv
+import requests
+import time
+import datatable
+from typing import List
+
+# Pump data to MLOps
+
+# X is your input dt.Frame (https://github.com/h2oai/datatable)
+X
+
+# This endpoint will need to be updated by copying your endpoint URL from current deployed model 
+scoring_endpoint = "http://34.217.40.39:1080/85bfcf27-fbbb-4248-aa1d-49796c4bc328/model/score"
+
+num_pumps = 1
+
+X_pd = X.to_pandas()
+
+for i in range(num_pumps):
+    for row in range(X.nrows):
+        print(list(X.names))
+        print(list(list(X_pd.iloc[row].values)))
+        response = requests.post(
+            url = scoring_endpoint,
+            json = {"fields": list(X.names), "rows": [[str(num) for num in list(X_pd.iloc[0].values)]]},
+        )
+        time.sleep(1)
+
+return X  # return dt.Frame, pd.DataFrame, np.ndarray or a list or named dict of those
+~~~
+
+![dai-recipe-live-code-data-pumper](./assets/dai-recipe-live-code-data-pumper.jpg)
+
+Alternatively, you could run a Python script from your CLI to execute the data pumper. Refer to the script file: [data-pumper-called-from-cli.py](./recipes/data-pumper-called-from-cli.py)
```

Comment:
So it looks like there are 3 ways to pump in data:

1. Driverless AI pump as described in this section.
2. The offline CLI with help from this linked script.
3. Using the data pump built into the Platform Studio web app.

My 2 cents:
- The data pump build into Studio page is **useful** because it's easy.  They just drop in a URL and things work.
- The offline script is **useful** because someone can see sort of plainly what the HTTP calls look like.
- The DAI approach is **somewhat confusing**.  The code non-trivial (transforming frames from something named `X` and unclear what the schema looks like).  I get that it's a useful feature, but I wonder if it adds any value to an **MLOps** tutorial.

Just sharing my thoughts - not pushing either way.

Reply:
@orendain I do remember the studio app being very user friendly, but with the current version of MLOps test drive that I have, the data pump only supports the credit card dataset. So, I created a live code data recipe to act as a data pump. It was sort of a hack since I don't believe data recipe was meant to do that. I decided not to use python from cli since the user would be pumping data outside the environment of DAI and MLOps. It would be nice to have live code python editor in MLOps like DAI.

Reply:
@orendain for now I will keep the DAI approach to pump data to MLOps. But I would prefer to use the studio app or if MLOps had a built in live code python editor, then that would also be a good potential solution.